sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubClient.java

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub.client;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.util.BinaryData;
+import com.azure.messaging.webpubsub.client.implementation.EventHandlerCollection;
+import com.azure.messaging.webpubsub.client.implementation.WebPubSubClientState;
+import com.azure.messaging.webpubsub.client.implementation.websocket.WebSocketSession;
+import com.azure.messaging.webpubsub.client.models.ConnectFailedException;
+import com.azure.messaging.webpubsub.client.models.ConnectedEvent;
+import com.azure.messaging.webpubsub.client.models.DisconnectedEvent;
+import com.azure.messaging.webpubsub.client.models.GroupMessageEvent;
+import com.azure.messaging.webpubsub.client.models.RejoinGroupFailedEvent;
+import com.azure.messaging.webpubsub.client.models.SendEventOptions;
+import com.azure.messaging.webpubsub.client.models.SendMessageFailedException;
+import com.azure.messaging.webpubsub.client.models.SendToGroupOptions;
+import com.azure.messaging.webpubsub.client.models.ServerMessageEvent;
+import com.azure.messaging.webpubsub.client.models.StoppedEvent;
+import com.azure.messaging.webpubsub.client.models.WebPubSubDataType;
+import com.azure.messaging.webpubsub.client.models.WebPubSubResult;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.function.Consumer;
+
+/**
+ * The WebPubSubAsync client.
```

Comment:
This should be the sync client and also, this is the public client, we should have more detailed javadoc.